src/v/cluster/metadata_cache.cc

```diff
@@ -272,6 +280,11 @@ topic_properties metadata_cache::get_default_properties() const {
     tp.recovery = {false};
     tp.shadow_indexing = {get_default_shadow_indexing_mode()};
     tp.batch_max_bytes = get_default_batch_max_bytes();
+    tp.retention_local_target_bytes = tristate{
+      get_default_retention_local_target_bytes()};
+    tp.retention_local_target_ms = tristate<std::chrono::milliseconds>{
```

Comment:
question: does this really need to be a tristate? looks like it's always initialized to some value

Reply:
I think they have to be tristates because we initialise them from `retention_bytes` and `retention_ms` on upgrades (see [here](https://github.com/redpanda-data/redpanda/pull/6613/files#diff-01886c33d71d07cec65471ac49816a36812ebae6ba157847e350e7a944356f4aR137-R138)).